src/pip/_internal/cli/req_command.py

```diff
@@ -158,6 +160,30 @@ def handle_pip_version_check(self, options):
 ]
 
 
+def warn_if_run_as_root():
+    # type: () -> None
+    """Output a warning for sudo users on Unix.
+
+    In a virtual environment, sudo pip still writes to virtualenv.
+    On Windows, users may run pip as Administrator without issues.
+    This warning only applies to Unix root users outside of virtualenv.
+    """
+    if running_under_virtualenv():
+        return
+    if not hasattr(os, "getuid"):
+        return
+    if sys.platform == "win32" or sys.platform == "cygwin":
+        return
+    elif sys.platform == "darwin" or sys.platform == "linux":
+        if os.getuid() != 0:
```

Comment:
I'd flip the parent conditional and have an early return.

Reply:
Unless mypy gets angry. :(

Reply:
I've tried pretty much every other format...mypy only likes this one :/